src/hypothesis/extra/django/models.py

```diff
@@ -166,25 +161,22 @@ def _get_strategy_for_field(f):
 
 def models(model, **extra):
     result = {}
-    mandatory = set()
+    missed = set()
     for f in model._meta.concrete_fields:
-        try:
-            strategy = _get_strategy_for_field(f)
-        except UnmappedFieldError:
-            mandatory.add(f.name)
-            continue
-        if strategy is not None:
+        strategy = _get_strategy_for_field(f)
+        if isinstance(strategy, st.SearchStrategy):
             result[f.name] = strategy
-    missed = {x for x in mandatory if x not in extra}
+        elif strategy is None and not (f.name in extra or f.null):
+            missed.add(f.name)
     if missed:
         raise InvalidArgument((
             u'Missing arguments for mandatory field%s %s for model %s' % (
                 u's' if len(missed) > 1 else u'',
-                u', '.join(missed),
+                u', '.join(sorted(missed)),
                 model.__name__,
             )))
     result.update(extra)
-    # Remove default_values so we don't try to generate anything for those.
+    # Remove default_values, for backwards-compatibility
```

Comment:
I don't understand this comment. What's the change in behaviour here and what is the backwards-compatibility issue?